ipapython/ipautil.py

```diff
@@ -515,14 +515,18 @@ def run(args, stdin=None, raiseonerr=True, nolog=(), env=None,
             for group, gid in zip(suplementary_groups, suplementary_gids):
                 logger.debug('suplementary_group=%s (GID %d)', group, gid)
 
-    def preexec_fn():
-        if runas is not None:
-            os.setgroups(suplementary_gids)
-            os.setregid(pent.pw_gid, pent.pw_gid)
-            os.setreuid(pent.pw_uid, pent.pw_uid)
-
-        if umask:
-            os.umask(umask)
+    if runas is not None or umask is not None:
+        # preexec function is not supported in WSGI environment
+        def preexec_fn():
+            if runas is not None:
+                os.setgroups(suplementary_gids)
```

Comment:
also worth noting: in Python 3.9 the subprocess module gains support for doing this without preexec_fn:
  https://bugs.python.org/issue36046

i don't believe we have umask support yet, but that one is easier to work around via a child process shell that calls umask before exec'ing the target.  For anything you find a preexec_fn being used for, feel free to file subprocess feature requests with upstream CPython.